wolfcrypt/src/port/nxp/se050_port.c

```diff
@@ -1268,4 +1269,168 @@ int se050_ed25519_verify_msg(const byte* signature, word32 signatureLen,
+    size_t keyPairExportLen = sizeof(keyPairExport);
+    size_t keyPairExportBitLen = sizeof(keyPairExport) * 8;
+    int ret;
+
+    (void)keySize;
+
+    if (cfg_se050_i2c_pi == NULL) {
+        return WC_HW_E;
+    }
+
+    if (wolfSSL_CryptHwMutexLock() != 0) {
+        return BAD_MUTEX_E;
+    }
+    status = sss_key_store_context_init(&host_keystore, cfg_se050_i2c_pi);
+    if (status == kStatus_SSS_Success) {
+        status = sss_key_store_allocate(&host_keystore, 60);
+    }
+
+    if (status == kStatus_SSS_Success) {
+        status = sss_key_object_init(&keyPair, &host_keystore);
+    }
+
+    if (status == kStatus_SSS_Success) {
+        status = sss_key_object_allocate_handle(&keyPair, keyId,
+            kSSS_KeyPart_Pair, kSSS_CipherType_EC_MONTGOMERY, 256,
+            kKeyObject_Mode_None);
+    }
+
+    if (status == kStatus_SSS_Success) {
+        status = sss_key_store_generate_key(&host_keystore, &keyPair,
+            256, NULL);
+    }
+
+    if (status == kStatus_SSS_Success) {
+        status = sss_key_store_get_key(&host_keystore, &keyPair,
+            keyPairExport, &keyPairExportLen, &keyPairExportBitLen);
+    }
+
+    wolfSSL_CryptHwMutexUnLock();
+    if (status == kStatus_SSS_Success) {
+        key->keyId = keyId;
+        ret = 0;
+    }
+
+    else {
+        ret = WC_HW_E;
+    }
+    return ret;
+}
+
+
+int se050_curve25519_shared_secret(curve25519_key* private_key, curve25519_key* public_key,
+    byte* out, word32* outlen)
+{
+    sss_status_t            status = kStatus_SSS_Success;
+    sss_key_store_t         host_keystore;
+    sss_key_store_t         host_keystore_2;
+    sss_object_t            ref_private_key;
+    sss_object_t            ref_public_key;
+    sss_object_t            deriveKey;
+    sss_derive_key_t        ctx_derive_key;
+    int                     keyId;
+    int                     keySize = CURVE25519_KEYSIZE;
+    size_t                  ecdhKeyLen = keySize;
+    size_t                  ecdhKeyBitLen = keySize;
+    int                     ret = WC_HW_E;
+
+    if (cfg_se050_i2c_pi == NULL) {
+        return WC_HW_E;
+    }
+
+    if (private_key->keyId <= 0 || public_key->keyId <= 0) {
+        return BAD_FUNC_ARG;
+    }
+
+    if (wolfSSL_CryptHwMutexLock() != 0) {
+        return BAD_MUTEX_E;
+    }
+
+    status = sss_key_store_context_init(&host_keystore, cfg_se050_i2c_pi);
+    if (status == kStatus_SSS_Success) {
+        status = sss_key_store_allocate(&host_keystore, 60);
+    }
+
+    if (status == kStatus_SSS_Success) {
+        status = sss_key_object_init(&ref_public_key, &host_keystore);
+    }
+
+    if (status == kStatus_SSS_Success) {
+        status = sss_key_object_get_handle(&ref_public_key, public_key->keyId);
+    }
+
+    if (status == kStatus_SSS_Success) {
+        status = sss_key_store_context_init(&host_keystore_2, cfg_se050_i2c_pi);
+    }
+
+    if (status == kStatus_SSS_Success) {
+        status = sss_key_store_allocate(&host_keystore_2, 60);
```

Comment:
Same. Use `SE050_KEYSTOREID_ED25519`. Or add a new one for `SE050_KEYSTOREID_CURVE25519` that is different.